flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/tasks/SubtaskCheckpointCoordinatorImpl.java

```diff
@@ -65,6 +75,8 @@ public void checkpointState(
 			CheckpointMetrics checkpointMetrics,
 			OperatorChain<?, ?> operatorChain,
 			Supplier<Boolean> isCanceled) throws Exception {
+		checkNotNull(checkpointOptions);
```

Comment:
nit: also `checkNotNull` for `checkpointMetaData` and `operatorChain`, and i think they can be done with the commit `[FLINK-16744][task][refactor] extract SubtaskCheckpointCoordinator`

Reply:
Both of these fields are unreferenced a few lines below. Do you think it makes sense to add this check at the beginning too?

Reply:
When I checked the commit "[FLINK-16744][task][refactor] extract SubtaskCheckpointCoordinator",  the arguments of `CheckpointMetaData`, `CheckpointOptions` and `operatorChain` are already referenced in below lines, so i suggested `checkNotNull` for them in that commit.

Reply:
When checking the commit "[FLINK-16744][task][refactor] inline CheckpointingOperation", we only `checkNotNull` for `checkpointOptions` and `checkpointMetrics`.  But from the commit "[FLINK-16744][task][refactor] extract SubtaskCheckpointCoordinator" when this method is firstly introduced,  we did not `checkNotNull` for other arguments which are actually referenced in that commit, so the criteria seems inconsistent.